akka-javasdk/src/main/scala/akka/javasdk/impl/agent/FunctionTools.scala

```diff
@@ -54,34 +73,64 @@ object FunctionTools {
   private def collectFunctionToolInvokers(cls: Class[_], allowNonPublic: Boolean)(
       instanceFactory: () => Any): Map[String, FunctionToolInvoker] = {
 
-    cls.getDeclaredMethods
-      .filter(m => m.hasAnnotation[FunctionTool])
-      .filter(m => m.isPublic || allowNonPublic)
-      .map { method =>
-
-        val toolAnno = method.getAnnotation(classOf[FunctionTool])
-        val name =
-          if (toolAnno.name() == null || toolAnno.name().isBlank) method.getName
-          else toolAnno.name()
+    annotatedMethods(cls, allowNonPublic).map { method =>
 
-        name ->
-        new FunctionToolInvoker {
-          override def paramNames: Array[String] =
-            method.getParameters.map(_.getName)
+      val name = toolName(method)
 
-          override def types: Array[Type] =
-            method.getGenericParameterTypes
+      name ->
+      new FunctionToolInvoker {
+        override def paramNames: Array[String] =
+          method.getParameters.map(_.getName)
 
-          override def invoke(args: Array[Any]): Any = {
-            if (allowNonPublic) method.setAccessible(true)
-            val instance = instanceFactory()
-            method.invoke(instance, args: _*)
-          }
+        override def types: Array[Type] =
+          method.getGenericParameterTypes
 
-          override def returnType: Class[_] =
-            method.getReturnType
+        override def invoke(args: Array[Any]): Any = {
+          if (allowNonPublic) method.setAccessible(true)
+          val instance = instanceFactory()
+          method.invoke(instance, args: _*)
         }
+
+        override def returnType: Class[_] =
+          method.getReturnType
       }
-      .toMap
+    }.toMap
+  }
+
+  private def toolName(method: Method): String = {
+    val toolAnno = method.getAnnotation(classOf[FunctionTool])
+
+    if (toolAnno.name() == null || toolAnno.name().isBlank) method.getName
+    else toolAnno.name()
+
+  }
```

Comment:
This is in preparation for the next PR in which I will ensure that functions names are unique.